src/NexusMods.DataModel/FileHashCache.cs

```diff
@@ -34,44 +30,46 @@ public class FileHashCache : IFileHashCache
     public static readonly ActivityGroup Group = ActivityGroup.From("FileHashCache");
 
     private readonly IActivityFactory _activityFactory;
-    private readonly IDataStore _store;
+    private readonly IConnection _conn;
 
     /// <summary/>
     /// <param name="activityFactory">Limits CPU utilization where possible.</param>
     /// <param name="store">The store inside which the file hashes are kept within.</param>
     /// <remarks>
     ///    This constructor is usually called from DI.
     /// </remarks>
-    public FileHashCache(IActivityFactory activityFactory, IDataStore store)
+    public FileHashCache(IActivityFactory activityFactory, IConnection conn)
     {
         _activityFactory = activityFactory;
-        _store = store;
+        _conn = conn;
     }
 
     /// <inheritdoc/>
-    public bool TryGetCached(AbsolutePath path, out FileHashCacheEntry entry)
+    public bool TryGetCached(AbsolutePath path, out HashCacheEntry.Model entry)
     {
-        var normalized = path.ToString();
-        Span<byte> span = stackalloc byte[Encoding.UTF8.GetByteCount(normalized)];
-        Encoding.UTF8.GetBytes(normalized, span);
-        var found = _store.GetRaw(IId.FromSpan(EntityCategory.FileHashes, span));
-        if (found != null && found is not { Length: 0 })
+        var nameHash = path.ToString().XxHash64AsUtf8();
```

Comment:
Copy this class out of my branch and use it:

https://github.com/Nexus-Mods/NexusMods.App/blob/93f53060927f1a693b7c0ee7da59cd6972791409/src/Extensions/NexusMods.Extensions.Hashing/StringExtensions.cs#L6-L28

It should be a bit faster, since there's no string conversion done. Might speed up things a bit since this can be hit on `IndexFoldersAsync`, a hot path.

[Don't forget about the corresponding write!]